src/main/java/enmasse/kafka/bridge/Bridge.java

```diff
@@ -396,4 +404,46 @@ private void processOpenSender(ProtonConnection connection, ProtonSender sender)
 
 		sink.handle(sender);
 	}
+	
+	static ErrorCondition newError(String error, String description) {
+		return new ErrorCondition(Symbol.getSymbol(error), description);
+	}
+	
+	static void detachWithError(ProtonLink<?> link, String error, String description) {
+		detachWithError(link, newError(error, description));
+	}
+
+	static void detachWithError(ProtonLink<?> link, ErrorCondition error) {
+		LOG.error("Detaching link {} due to eror {}, description: {}", link, error.getCondition(), error.getDescription());
+		link.setSource(null)
+		.open()
+		.setCondition(error)
+		.close();
+	}
+	
+	static MessageConverter<?, ?> instantiateConverter(ProtonLink<?> link, String className) {
```

Comment:
aren't we doing too much responsibilities to this method related to send the error on AMQP as well ? Maybe the endpoint sink/source which uses this method for converter creation can check and send AMQP error to its client counter part on the network. Wdyt ?